### PR TITLE
Remove Redundant Functions + 60% MD Speedup

### PR DESCRIPTION
**Pull request by GNiendorf**
Below is run on the L40.

This PR Timing:
Screenshot 2024-09-26 at 4 13 20 PM
Master Timing:
Screenshot 2024-09-26 at 4 15 43 PM

MD creation time goes from 1.0ms to 0.4ms, meaning that the redundant matrix creation mentioned below represented the majority of the MD creation time on GPU... 😱 @slava77

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h [hunk 1]
```diff
@@ -283,35 +283,6 @@ namespace lst {
   };
 
   ALPAKA_FN_ACC ALPAKA_FN_INLINE float moduleGapSize(struct lst::Modules const& modulesInGPU, uint16_t moduleIndex) {
-    float miniDeltaTilted[3] = {0.26f, 0.26f, 0.26f};
-    float miniDeltaFlat[6] = {0.26f, 0.16f, 0.16f, 0.18f, 0.18f, 0.18f};
-    float miniDeltaLooseTilted[3] = {0.4f, 0.4f, 0.4f};
-    float miniDeltaEndcap[5][15];
-
-    for (size_t i = 0; i < 5; i++) {
```
Comment (GNiendorf) [inline RecoTracker/LSTCore/src/alpaka/MiniDoublet.h:291]: This was being run many times, representing a significant portion of the MD creation time it seems like.

**Comment by slava77** [comment]
> MD creation time goes from 1.0ms to 0.4ms, meaning that the redundant matrix creation mentioned below represented the majority of the MD creation time on GPU

nice find/fix.
Thank you.

**Comment by slava77** [comment]
apparently this is addressing https://github.com/SegmentLinking/TrackLooper/issues/303

**Comment by VourMa** [comment]
Indeed, nice catch and clean up. I would even considering adding this to the CMSSW PR (especially if the renaming to adhere to CMSSW naming is real and not an artefact of the rebases). What do you think?

@ariostas Is the `_devel` branch fully up to date with the branch for the CMSSW PR?

**Comment by ariostas** [comment]
The `_devel` branch is not up to date since there are still some big changes coming. I actually just noticed that the `_devel` PR is a bit of a mess, so we'll have to figure that out at some point. But I agree that this should go into the `_realfiles` branch. This is great!

I think there's other PRs by Gavin that should also be included in `_realfiles` since there have been review comments about them. In particular, #39 and #54.

**Comment by VourMa** [comment]
> I think there's other PRs by Gavin that should also be included in _realfiles since there have been review comments about them. In particular, https://github.com/SegmentLinking/cmssw/pull/39 and https://github.com/SegmentLinking/cmssw/pull/54.

I would be less eager to also push these ones, because:

1. They introduce a new (physics) feature (lower pT cut) which is not strictly needed at this stage.
2. They require new data files, with all the complication this brings to the cms-sw way of running things for PRs.
3. They define a clear-cut development, separated from the rest (up to a good degree).

So I think they would be great for a standalone PR.

**Comment by ariostas** [comment]
> I would be less eager to also push these ones, because...

That's true. But Andrea has asked for occupancy tables instead of a bunch of if-else statements, and if we want to fix SDL_INF we'll have to update the data files anyway, so maybe we could just get that out of the way

**Comment by slava77** [comment]
linter check isn't happy here either

**Comment by GNiendorf** [comment]
@ariostas When I run the code format I see a bunch of suggestions for files not related to this PR:

Screenshot 2024-09-27 at 2 46 24 PM

**Comment by slava77** [comment]
> @ariostas When I run the code format I see a bunch of suggestions for files not related to this PR:

did you just update to the new release or still reused the old release area with a new cmsrel?

**Comment by GNiendorf** [comment]
Oh no, I haven't updated. Let me do that and try again.

**Comment by ariostas** [comment]
> When I run the code format I see a bunch of suggestions for files not related to this PR:

I have the same issue. I'm not sure how to get around it. I think they didn't format all files after clang-format got updated

**Review by slava77 — APPROVED** [review]

**Comment by slava77** [comment]
the linter complaints are for parts of the files not touched by this PR

**Comment by slava77** [comment]
> Here is a timing comparison:

there is practically no change in the CPU backend: MD changes from 326.4 to 323.1; the other run was faster by around 1 s.
I guess the optimization works out differently (or something else is very slow)

**Pull request merged**